gateleen-core/src/main/java/org/swisspush/gateleen/core/util/HttpHeaderUtil.java

```diff
@@ -61,18 +68,43 @@
      * Helper method to check the presence of a http header matching the provided pattern. Every header entry is checked
      * in the format KEY: VALUE
      *
-     * @param headers   The Map with the header key - value pairs to be evaluated
+     * @param headers        The Map with the header key - value pairs to be evaluated
      * @param headersPattern The pattern to match the headers against.
      * @return True if a matching header was found, false otherwise
      */
     public static <T extends MultiMap> boolean hasMatchingHeader(@Nonnull T headers, @Nonnull Pattern headersPattern) {
         for (Map.Entry<String, String> entry : headers) {
             String header = entry.getKey() + ": " + entry.getValue();
-            if(headersPattern.matcher(header).matches()){
+            if (headersPattern.matcher(header).matches()) {
                 return true;
             }
         }
         return false;
     }
 
+    /**
+     * Merges headers, makes sure that only one value of header {@link HttpRequestHeader#CONTENT_LENGTH} ends up in the result.
+     */
+    public static void mergeHeaders(@Nonnull MultiMap destination, @Nonnull MultiMap source, @Nullable String context) {
+        String destinationContentLength = "";
+        String sourceContentLength = "";
+
+        if (HttpRequestHeader.containsHeader(destination, HttpRequestHeader.CONTENT_LENGTH)) {
+            destinationContentLength = destination.get(HttpRequestHeader.CONTENT_LENGTH.getName());
+        }
```

Comment:
(*trivial*) [Doc of MultiMap.get(String)](https://vertx.io/docs/apidocs/io/vertx/core/MultiMap.html#get-java.lang.String-) says:
> Returns: The first header value **or null if there is no such entry**

I guess could also be written without the `contains` check:
```java
    destinationContentLength = destination.get(HttpRequestHeader.CONTENT_LENGTH.getName());
    if( destinationContentLength == null ){
        destinationContentLength = "";
    }
```
Just an idea. I think its not important how we do it.

Reply:
Yes, but the readabiltity is better with "contains" I would say. Implemention wise we end up in a get() in both cases